src/commands/selectCommandTemplate.ts

```diff
@@ -47,9 +47,9 @@ export async function selectRunCommand(context: IActionContext, fullTag: string,
     return await selectCommandTemplate(
         context,
         interactive ? 'runInteractive' : 'run',
-        [fullTag],
+        [imageId],
         undefined,
-        { 'tag': fullTag, 'exposedPorts': portsString, 'containerCommand': await ext.runtimeManager.getCommand() }
+        { 'imageId': imageId, 'exposedPorts': portsString, 'containerCommand': await ext.runtimeManager.getCommand() }

```

Comment:
`${tag}` may be in use by non-default command configurations, so we can't get rid of it. We can add `imageId` (or `tagOrImageId`) to this list, however. The value can be the image name if it is named, or the ID if it is not (and the name is `<none>` or `<none>:<none>`, I can't recall)

Reply:
Okay, i agree with you.
Change back tag field. And the value can be the image name or id:

![image](https://github.com/microsoft/vscode-docker/assets/8070232/e7ebbcb4-c51d-4e25-a52f-dff9fc2bb78e)